src/main/java/com/google/cloud/spanner/pgadapter/utils/Metrics.java

```diff
@@ -0,0 +1,75 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.cloud.spanner.pgadapter.utils;
+
+import com.google.api.core.InternalApi;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.LongHistogram;
+import io.opentelemetry.api.metrics.Meter;
+import java.util.Arrays;
+import java.util.List;
+
+@InternalApi
+public class Metrics {
+  static final String INSTRUMENTATION_SCOPE = "cloud.google.com/java";
+  static final String SPANNER_CLIENT_LIB_LATENCY = "spanner/pgadapter/client_lib_latencies";
+  static final String SPANNER_CLIENT_LIB_LATENCY_DESCRIPTION =
+      "Latency when the client library receives a call and returns a response";
+  static final String PGADAPTER_LATENCY = "spanner/pgadapter/roundtrip_latencies";
+  static final String PGADAPTER_LATENCY_DESCRIPTION =
+      "Latency when the JDBC makes a call and gets a response, measured in PGAdapter";
+
+  private final LongHistogram spannerClientLibLatencies;
+  private final LongHistogram pgadapterLatencies;
+
+  public Metrics(OpenTelemetry openTelemetry) {
+    Meter meter = openTelemetry.getMeter(INSTRUMENTATION_SCOPE);
+    List<Long> RPC_MILLIS_BUCKET_BOUNDARIES =
+        Arrays.asList(
+            1L, 2L, 3L, 4L, 5L, 6L, 8L, 10L, 13L, 16L, 20L, 25L, 30L, 40L, 50L, 65L, 80L, 100L,
+            130L, 160L, 200L, 250L, 300L, 400L, 500L, 650L, 800L, 1000L, 2000L, 5000L, 10000L,
+            20000L, 50000L, 100000L);
+    spannerClientLibLatencies =
+        meter
+            .histogramBuilder(SPANNER_CLIENT_LIB_LATENCY)
+            .ofLongs()
+            .setDescription(SPANNER_CLIENT_LIB_LATENCY_DESCRIPTION)
+            .setUnit("ms")
+            .setExplicitBucketBoundariesAdvice(RPC_MILLIS_BUCKET_BOUNDARIES)
+            .build();
+    pgadapterLatencies =
+        meter
+            .histogramBuilder(PGADAPTER_LATENCY)
+            .ofLongs()
+            .setDescription(PGADAPTER_LATENCY_DESCRIPTION)
+            .setUnit("ms")
+            .setExplicitBucketBoundariesAdvice(RPC_MILLIS_BUCKET_BOUNDARIES)
+            .build();
+  }
+
+  @InternalApi
+  public void recordClientLibLatency(long value, Attributes attributes) {
+    if (spannerClientLibLatencies != null) {
+      spannerClientLibLatencies.record(value, attributes);
+    }
+  }
+
+  @InternalApi
+  public void recordPGAdapterLatency(long value, Attributes attributes) {
+    if (pgadapterLatencies != null) {
```

Comment:
Is this ever null?

And if it isn't (which based on the above code I would guess that it never is), could we have some other flag that indicates that a quick return is ok. For example some kind of flag that says 'metrics are disabled'.

Reply:
I got this from the [java client lib](https://github.com/googleapis/java-spanner/blob/4f7d28bdbabce5faf6d37b8a3224235bb1012f78/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerRpcMetrics.java#L61-L73). 

I will remove these null checks and add the support of a flag to enable/disable metrics.